test/E2ETest/Microsoft.Test.E2E.AspNet.OData/DollarLevels/DollarLevelsTest.cs

```diff
@@ -56,15 +56,13 @@ public async Task LevelsWithInvalidValue(string query, string errorMessage)
         }
 
         [Theory]
-        [InlineData("$expand=Manager($levels=3)", 
-            "$expand=Manager($expand=Manager($expand=Manager))")]
-        [InlineData("$expand=Manager($levels=0)", 
-            "")]
+        [InlineData("$expand=Manager($levels=3)", "$expand=Manager($expand=Manager($expand=Manager))")]
+        [InlineData("$expand=Manager($levels=0)", "")]
         [InlineData("$expand=Manager($levels=2;$expand=DirectReports($levels=2))",
             "$expand=Manager($expand=Manager($expand=DirectReports($expand=DirectReports)),DirectReports($expand=DirectReports))")]
         [InlineData("$expand=Manager($select=ID;$expand=DirectReports($levels=2;$select=Name))",
-            "$expand=Manager($select=ID;$expand=DirectReports($expand=DirectReports($select=Name);$select=Name))")]
-        public async Task LevelsWithValidNumber(string originalQuery, string expandedQuery) 
+            "$expand=Manager($select=ID;$expand=DirectReports($expand=DirectReports($select=Name);$select=Name,DirectReports))")]
```

Comment:
>DirectReports [](start = 111, length = 13)

Why is this additional DirectReports appended? The original query does not select the DirectReports.

Reply:
It looks required so far from the logical of ProcessLevel(..).